spark/interpreter/src/main/java/org/apache/zeppelin/spark/OldSparkInterpreter.java

```diff
@@ -1,1281 +0,0 @@
-      try { // for RcpEnv
-        Method getClassOutputDirectory = intp.getClass().getMethod("getClassOutputDirectory");
-        File classOutputDirectory = (File) getClassOutputDirectory.invoke(intp);
-        replClassOutputDirectory = classOutputDirectory.getAbsolutePath();
-      } catch (NoSuchMethodException | SecurityException | IllegalAccessException
-          | IllegalArgumentException | InvocationTargetException e) {
-        // continue
-      }
-    }
-
-    if (Utils.isScala2_11()) {
-      classServer = createHttpServer(outputDir);
-      Utils.invokeMethod(classServer, "start");
-      classServerUri = (String) Utils.invokeMethod(classServer, "uri");
-    }
-
-    if (classServerUri != null) {
-      conf.set("spark.repl.class.uri", classServerUri);
-    }
-
-    if (replClassOutputDirectory != null) {
-      conf.set("spark.repl.class.outputDir", replClassOutputDirectory);
-    }
-
-    if (jars.length > 0) {
-      conf.setJars(jars);
-    }
-
-    if (execUri != null) {
-      conf.set("spark.executor.uri", execUri);
-    }
-    conf.set("spark.scheduler.mode", "FAIR");
-
-    Properties intpProperty = getProperties();
-    for (Object k : intpProperty.keySet()) {
-      String key = (String) k;
-      String val = toString(intpProperty.get(key));
-      if (!val.trim().isEmpty()) {
-        if (key.startsWith("spark.")) {
-          logger.debug(String.format("SparkConf: key = [%s], value = [%s]", key, val));
-          conf.set(key, val);
-        }
-
-        if (key.startsWith("zeppelin.spark.")) {
-          String sparkPropertyKey = key.substring("zeppelin.spark.".length());
-          logger.debug(String.format("SparkConf: key = [%s], value = [%s]", sparkPropertyKey, val));
-          conf.set(sparkPropertyKey, val);
-        }
-      }
-    }
-    SparkContext sparkContext = new SparkContext(conf);
-    return sparkContext;
-  }
-
-  static final String toString(Object o) {
-    return (o instanceof String) ? (String) o : "";
-  }
-
-  public static boolean useSparkSubmit() {
-    return null != System.getenv("SPARK_SUBMIT");
-  }
-
-  public boolean printREPLOutput() {
-    return java.lang.Boolean.parseBoolean(getProperty("zeppelin.spark.printREPLOutput"));
-  }
-
-  @Override
-  public void open() throws InterpreterException {
-    this.enableSupportedVersionCheck = java.lang.Boolean.parseBoolean(
-        getProperty("zeppelin.spark.enableSupportedVersionCheck", "true"));
-
-    // set properties and do login before creating any spark stuff for secured cluster
-    if (isYarnMode()) {
-      System.setProperty("SPARK_YARN_MODE", "true");
-    }
-    if (getProperties().containsKey("spark.yarn.keytab") &&
-        getProperties().containsKey("spark.yarn.principal")) {
-      try {
-        String keytab = getProperties().getProperty("spark.yarn.keytab");
-        String principal = getProperties().getProperty("spark.yarn.principal");
-        UserGroupInformation.loginUserFromKeytab(principal, keytab);
```

Comment:
Set the kerberos authentication information according to the configuration in the new spark interpreter.
Can it be added to the new spark interpreter?
This is very useful.

Reply:
This is legacy code for OldSparkInterpreter. At that time, we didn't pass spark conf via `--conf` in spark-submit. But now, we correct that in SparkInterpreterLauncher, so we don't need to do that again.